topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinator.java

```diff
@@ -10,33 +10,41 @@
 import io.camunda.zeebe.scheduler.future.ActorFuture;
 import io.camunda.zeebe.topology.state.ClusterTopology;
 import io.camunda.zeebe.topology.state.TopologyChangeOperation;
+import io.camunda.zeebe.util.Either;
 import java.util.List;
 
 public interface TopologyChangeCoordinator {
 
   /**
-   * Applies the given operations to the current cluster topology. The operations are applied
-   * asynchronously. The future is completed when the topology change has started successfully, but
-   * before completing. To query the status of the topology change, use {@link #hasCompletedChanges}
-   * with the version of the returned ClusterTopology.
+   * Applies the operations generated by the requestTransformer to the current cluster topology. If
+   * no operations is returned by requestTransformer, the future completes successfully with no
+   * change to the cluster topology.
    *
-   * <pre>
-   *  final ActorFuture<ClusterTopology> topologyWithPendingChanges =
-   *         coordinator.applyOperations(operations).join();
-   *  final var topologyChangeIsCompleted =
-   *         coordinator.hasCompletedChanges(topologyWithPendingChanges.version()).join();
-   * </pre>
-   *
-   * @param operations the operations to apply
-   * @return a future which is completed with the intermediate topology with pending operations
+   * @param requestTransformer the request transformer that generates the operations to apply
+   * @return a future which is completed when the topology change has started successfully.
    */
-  ActorFuture<ClusterTopology> applyOperations(List<TopologyChangeOperation> operations);
+  ActorFuture<TopologyChangeResult> applyOperations(TopologyChangeRequest requestTransformer);
```

Comment:
:upside_down_face: We could still support a plain list of operations. This would make tests easier and could also make sense for later APIs where we want full control over the operations and just want to provide a list.

Reply:
> We could still support a plain list of operations. 

This won't work because most request need to read current topology to generate the list of operations.